compiler/noirc_evaluator/src/ssa/opt/inlining/inline_info.rs

```diff
@@ -158,22 +160,37 @@ fn compute_function_should_be_inlined(
         return; // Already processed
     }
 
+    let function = &ssa.functions[&func_id];
+    let contains_static_assertion = function.reachable_blocks().iter().any(|block| {
+        function.dfg[*block].instructions().iter().any(|instruction| {
+            let assert_constant_id = function.dfg.get_intrinsic(Intrinsic::AssertConstant).copied();
+            let static_assert_id = function.dfg.get_intrinsic(Intrinsic::StaticAssert).copied();
```

Comment:
```suggestion
        let assert_constant_id = function.dfg.get_intrinsic(Intrinsic::AssertConstant).copied();
        let static_assert_id = function.dfg.get_intrinsic(Intrinsic::StaticAssert).copied();
        function.dfg[*block].instructions().iter().any(|instruction| {
```

Swapping the order of these so that we don't look up these function ids on every instruction (we could pull it out of the outer loop as well but :shrug:.

Reply:
I ended up doing this manually as we can pull it out of the outer loop. I was hoping rust would be smart enough to pull it out, but better to just do it.